example/src/CheckoutView.js

```diff
@@ -86,42 +86,50 @@ const CheckoutView = () => {
     }
   };
 
+  const { config, paymentMethods, onConfigChanged } = usePaymentMethods();
+
+  useEffect(() => { onConfigChanged(config) },
+    [paymentMethods == null],
+  );
+
   return (
-    <PaymentMethodsContext.Consumer>
-      {(context) => (
-        <SafeAreaView style={[backgroundStyle, { flex: 1 }]}>
-          <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
-
-          <View style={[styles.topContentView]}>
-            <Text>
-              {context.config.amount.value} {context.config.amount.currency}
-            </Text>
-            <Text>
-              Country:{' '}
-              {context.paymentMethods == null
-                ? '❗️'
-                : getFlagEmoji(context.config.countryCode)}
-            </Text>
-            <Button
-              title="Refresh Payment Methods"
-              onPress={() => context.onConfigChanged(context.config)}
-            />
-          </View>
-
-          <AdyenCheckout
-            config={context.config}
-            paymentMethods={context.paymentMethods}
-            onSubmit={ (payload, nativeComponent) => { didSubmit(payload, nativeComponent, context.config) }}
-            onProvide={didProvide}
-            onFail={didFail}
-            onComplete={didComplete}
-          >
-            <PaymentMethods />
-          </AdyenCheckout>
-        </SafeAreaView>
-      )}
-    </PaymentMethodsContext.Consumer>
+    <SafeAreaView style={[backgroundStyle, { flex: 1 }]}>
+      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
+
+      <PaymentMethodsView config={config} paymentMethods={paymentMethods} />
+
+      <AdyenCheckout
+        config={config}
+        paymentMethods={paymentMethods}
+        onSubmit={(payload, nativeComponent) => { didSubmit(payload, nativeComponent, config) }}
+        onProvide={didProvide}
+        onFail={didFail}
+        onComplete={didComplete}
+      >
+        <PaymentMethods />
+      </AdyenCheckout>
+    </SafeAreaView>
   );
 };
 
+const PaymentMethodsView = (props) => {
```

Comment:
You can destructure the props here, so the Component code would be less verbose:

```jsx
const PaymentMethodsView = ({ paymentMethods, config }) => { ...
```

Reply:
👍

Reply:
Fixed in [17aae14](https://github.com/Adyen/adyen-react-native/pull/56/commits/17aae14ac8766f76a157d7198033915d4d4426e6)